### PR TITLE
avoid no necessary mocha.run()

### PR DESCRIPTION
**Pull request by materliu**
Only tests run in real browser, if grunt-mocha options.run == true, the injected js by grunt-mocha will execute mocha.run() when everything has been done, add this if conditions to avoid no necessary mocha.run().

why I said the previous mocha.run() is not necessary. Because we all know that: 
`grunt-mocha injects a script into the PhantomJS instance that loads your HTML spec files. The file sets up a reporter and listeners so the output can be output in the command line. This option will call mocha.run() after the script is injected, ensuring that the proper listeners are setup.`

the injected script's position is below the previous mocha.run(), so when the mocha.run() is executed, the relative grunt listeners are not ready, on the console we will not receive the test report. eventually the injected js run mocha.run() again, which have us the console test case report.

### CODE REVIEW

File: generators/app/templates/index.html [hunk 1]
```diff
@@ -24,6 +24,17 @@
     <!-- include spec files here... -->
     <script src="spec/test.js"></script>
 
-    <script>mocha.run()</script>
+    <!-- run mocha after all test are loaded -->
+    <script>
+
```
Comment (arthurvr) [inline generators/app/templates/index.html]: ping @materliu

Comment (materliu) [inline generators/app/templates/index.html]: here @arthurvr

Comment (arthurvr) [inline generators/app/templates/index.html]: @materliu Can you do the updates necessary to this PR?

**Comment by arthurvr** [comment]
This will need a rebase as well.

**Comment by arthurvr** [comment]
ping @materliu

File: generators/app/templates/index.html (hunk 1 above)
Comment (materliu) [inline generators/app/templates/index.html]: I had, @arthurvr

Comment (arthurvr) [inline generators/app/templates/index.html]: @materliu You didn't do this one yet:

> Remove this line.

**Pull request closed without merging**